client/components/Categories/component.js

```diff
@@ -47,6 +45,22 @@ export default class CategoriesComponent extends Component {
     e.preventDefault();
 }
 
+  handleDelete = (e) => {
+    if(this.props.chosenCategoryId<0){
+      alert('You need to choose the category to delete!');
+    } else if (this.props.chosenCategoryId==this.props.categories.length-1){
+      alert('Category Other cannot be deleted');
+    } else {
+      const categories = this.props.categories;
+      $.post(`http://localhost:8000/categories/delete?categoryId=${categories[this.props.chosenCategoryId].categoryId}`).then(res => {
+        categories.splice(this.props.chosenCategoryId,1);
+        this.props.fetchCategories(categories);
+        this.props.chooseCategory(-1);
```

Comment:
To make colors of todos changed immediately ,we have to iterate through all todos and unset their categoryid on frontend as well.